### ISSUE TITLE
ThisType<T> isn't respected under the --strict master option

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#14521]
[Because the check uses `compilerOptions` instead of the `noImplicitThis` variable.](https://github.com/Microsoft/TypeScript/pull/14486/files#r104478900) 😉

Run the following example with `--strict`:

```ts
let x: ThisType<String> = {
    bar() {
        this.toLowerCase();
    }
}
```

**Expected:** No error
**Actual:** Implicit `any` error on `this`.

**Comment by ahejlsberg** [comment]
Already fixed in #14510.
